src/client/app_configuration_http.rs

```diff
@@ -61,26 +72,42 @@ impl<T: LiveConfiguration> ConfigurationProvider for AppConfigurationClientHttp<
     }
 
     fn get_feature(&self, feature_id: &str) -> Result<FeatureSnapshot> {
-        self.live_configuration.get_feature(feature_id)
+        let mut feature = self.live_configuration.get_feature(feature_id)?;
+        feature.metering = Some(self.metering.sender.clone());
+        Ok(feature)
```

Comment:
I don't like this way of _injecting_ the metering object into the `FeatureSnapshot` and `PropertySnapshot`, but I don't want to modify the public interface of these structs... I don't want outside users to know anything about metering.

I need something like this because only the `AppConfigurationClientHttp` implements metering. Maybe if we implement a noop metering, then we can do something different (better?).

In any case, if this is good enough, I would prefer to leave any alternate implementation to another PR (open issue with ideas).